### PR TITLE
[`pandas-vet`] autofixes for PD003, PD004, PD008, PD009 and PD011

### PR DESCRIPTION
**Pull request by sbrugman**
Autofixes for PD003, PD004, PD008, PD009 and PD011

### CODE REVIEW

File: src/rules/pandas_vet/rules/check_attr.rs [hunk 1]
```diff
@@ -95,7 +109,22 @@ pub fn check_attr(
...
 
-    checker
-        .diagnostics
-        .push(Diagnostic::new(violation, Range::from_located(attr_expr)));
+    let mut diagnostic = Diagnostic::new(violation, Range::from_located(attr_expr));
+    if checker.patch(diagnostic.kind.rule()) {
+        let replacement = match *diagnostic.kind.rule() {
+            Rule::UseOfDotAt => Some("loc"),
+            Rule::UseOfDotIat => Some("iloc"),
+            Rule::UseOfDotValues => Some("to_numpy()"),
```
Comment (spaceone) [inline src/rules/pandas_vet/rules/check_attr.rs:117]: I consider this fix dangerous as there is no check if `.values` is actually a pandas object.

Comment (sbrugman) [reply src/rules/pandas_vet/rules/check_attr.rs:117]: The fix is only applied when the use-of-dot-values is detected.

Comment (spaceone) [reply src/rules/pandas_vet/rules/check_attr.rs]: yes, and `use-of-dot-values` detects every `.values` access instead of only pandas related ones: #2229

Comment (sbrugman) [reply src/rules/pandas_vet/rules/check_attr.rs]: Yeah, thats partially limitation of not having type information. We could try to make the detection more conservative, but if the user enables it consciously, and the detection is correct, then the fix is in principle correct.. I think.

Comment (spaceone) [reply src/rules/pandas_vet/rules/check_attr.rs]: yes

Comment (charliermarsh) [reply src/rules/pandas_vet/rules/check_attr.rs]: My concern here is just that we still have known false-positives for this rule, and so we'll be knowingly breaking code in certain places by applying these autofixes. I'm tempted to say that these should be off-by-default, but enableable by users, which we haven't done before but I'd like to start doing in some cases.

Comment (sbrugman) [reply src/rules/pandas_vet/rules/check_attr.rs]: This one should benefit from the autofix safety levels (https://github.com/charliermarsh/ruff/issues/1992). Is there anything we could do in the meantime to merge this, like make the tests more conservative? Implemented this one for autofixing github.com/ing-bank/popmon

**Comment by sbrugman** [comment]
This PR probably needs autofix aggressiveness levels (https://github.com/charliermarsh/ruff/issues/1997)

**Comment by zanieb** [comment]
@sbrugman fwiw those levels are implemented now https://github.com/astral-sh/ruff/issues/4181 although they are not yet respected when running Ruff :)

**Comment by sbrugman** [comment]
Ok, sounds good! Anyone is free to pick up this where it was left off

**Pull request closed without merging**